File: app/src/main/java/com/ariefzuhri/myspecificnotification/ui/DetailsActivity.java
package com.ariefzuhri.myspecificnotification.ui;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;

import com.ariefzuhri.myspecificnotification.databinding.ActivityDetailsBinding;

import static com.ariefzuhri.myspecificnotification.utils.Constants.EXTRA_MESSAGE;

public class DetailsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityDetailsBinding binding = ActivityDetailsBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        Intent intent = getIntent();
        if (intent.hasExtra(EXTRA_MESSAGE)) {
            String extraMessage = intent.getStringExtra(EXTRA_MESSAGE);
            binding.textView.setText(extraMessage);
        }
    }
}

File: app/src/main/java/com/ariefzuhri/myspecificnotification/ui/DetailsActivity.kt
package com.ariefzuhri.myspecificnotification.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.ariefzuhri.myspecificnotification.databinding.ActivityDetailsBinding
import com.ariefzuhri.myspecificnotification.util.EXTRA_MESSAGE

class DetailsActivity : AppCompatActivity() {

    private lateinit var binding: ActivityDetailsBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityDetailsBinding.inflate(layoutInflater)
        setContentView(binding.root)

        showExtraMessage()
    }

    private fun showExtraMessage() {
        if (intent.hasExtra(EXTRA_MESSAGE)) {
            val extraMessage = intent.getStringExtra(EXTRA_MESSAGE)
            binding.textView.text = extraMessage
        }
    }
}